src/core.cpp

```diff
@@ -61,22 +118,60 @@ CVC::CVC(
         std::unique_ptr<DecisionEngine> decision_engine,
         std::vector<std::unique_ptr<Character>> characters,
         std::mt19937 random_generator) {
+    this->ticks_ = 0;
     this->random_generator_ = random_generator;
     this->decision_engine_ = std::move(decision_engine);
     this->characters_ = std::move(characters);
 }
 
+std::vector<Character *> CVC::GetCharacters() const {
+    std::vector<Character *> ret;//(this->characters_.size());
+    for(auto& character : this->characters_) {
+        ret.push_back(character.get());
```

Comment:
talking to @bhendrickson I should hang on to a vector of Character* and just handle the book keeping myself when characters go away. That is, I'll have one vector of std::unique_ptr<Character> that exists only to manage the lifecycle. This will become easier to understand when I better understand the lifecycle of Characters.

@bhendrickson also suggested I think about Character[] which would make iterating over all characters faster (because of memory locality). But then managing the lifecycle of one of those characters becomes more difficult.